src/App.js

```diff
@@ -38,6 +45,12 @@ class App extends React.Component {
   updateTodo = (id, completed) => {
     console.log("updateTodo", id, completed)
 
+    const todos = this.state.todos.map(todo => {
+      (todo.id === id) && (todo.completed = completed)
+      return todo
+    })
```

Comment:
Be careful when you're working with objects in state that you don't modify the objects directly; technically this still counts as 'mutating state' - on line 49 when you have `todo.completed = complete` this is mutating state directly. Instead, you should return a new todo object (which you can do with the spread operator) and then change its completed status:

However, in this case you *are* mutating state inside of the map function on line 49. Instead, you should create a new object inside the map function for the todo that you're updating, like this:

```js
    this.state.todos.map(todo => {
      if (todo.id === id) {
        return { ...todo, completed: completed } // spread makes a copy of the object, then we're updating the completed status of the new object
      }
      return todo
    })
```

Reply:
Shoot, you're right, that's a grave mistake!
Thanks for the feedback, Ian.